pkg/metricsclient/metricsclient.go

```diff
@@ -16,44 +16,51 @@ import (
 	"github.com/go-kit/kit/log/level"
 	"github.com/golang/snappy"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	clientmodel "github.com/prometheus/client_model/go"
 	"github.com/prometheus/common/expfmt"
 
+	"github.com/openshift/telemeter/pkg/prom"
 	"github.com/openshift/telemeter/pkg/reader"
 )
 
-var (
-	gaugeRequestRetrieve = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "metricsclient_request_retrieve",
-		Help: "Tracks the number of metrics retrievals",
-	}, []string{"client", "status_code"})
-	gaugeRequestSend = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "metricsclient_request_send",
-		Help: "Tracks the number of metrics sends",
-	}, []string{"client", "status_code"})
-)
+type Metrics struct {
+	gaugeRequestRetrieve *prometheus.GaugeVec
+	gaugeRequestSend     *prometheus.GaugeVec
+}
 
-func init() {
-	prometheus.MustRegister(
-		gaugeRequestRetrieve, gaugeRequestSend,
-	)
+func NewMetrics(reg prometheus.Registerer, metricsName string) *Metrics {
+	reg = prom.WrapRegistererWith(prometheus.Labels{"client": metricsName}, reg)
+	return &Metrics{
+		gaugeRequestRetrieve: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "metricsclient_request_retrieve",
+			Help: "Tracks the number of metrics retrievals",
+		}, []string{"status_code"}),
+		gaugeRequestSend: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "metricsclient_request_send",
+			Help: "Tracks the number of metrics sends",
+		}, []string{"status_code"}),
+	}
 }
 
 type Client struct {
-	client      *http.Client
-	maxBytes    int64
-	timeout     time.Duration
-	metricsName string
-	logger      log.Logger
+	logger  log.Logger
+	metrics *Metrics
+
+	client   *http.Client
+	maxBytes int64
+	timeout  time.Duration
 }
 
-func New(logger log.Logger, client *http.Client, maxBytes int64, timeout time.Duration, metricsName string) *Client {
+func New(logger log.Logger, metrics *Metrics, client *http.Client, maxBytes int64, timeout time.Duration) *Client {
```

Comment:
Same here, can we create the metrics directly inside of this creator func? And avoid having to expose another type "Metrics" and "NewMetrics"? That would simplify the package, the caller code etc. Also, it's really ambiguous what the `Metrics` type of the `metricsclient` package does.